engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1907,6 +1920,12 @@ public void updateRecoveryCheckpoints(PTOperator operator, UpdateCheckpointsCont
     if (operator.getState() == PTOperator.State.ACTIVE && (ctx.currentTms - operator.stats.lastWindowIdChangeTms) > operator.stats.windowProcessingTimeoutMillis) {
       // if the checkpoint is ahead, then it is not blocked but waiting for activation (state-less recovery, at-most-once)
       if (ctx.committedWindowId.longValue() >= operator.getRecoveryCheckpoint().windowId) {
+        LOG.info("Marking operator {} blocked: time since last windowId change={}, timeout={}, committed windowId={}, recovery windowId={}",
```

Comment:
How is this related to iteration support? Looks like some code was moved around.